elftools/dwarf/die.py

```diff
@@ -116,10 +118,26 @@ def get_full_path(self):
         fname = bytes2str(fname_attr.value) if fname_attr else ''
         return os.path.join(comp_dir, fname)
 
+    @lazy
+    def children(self):
+        """ Returns a tuple with all children of this DIE
+        """
+        # Note that lazy value is cached. Hence, it must be immutable.
+        return tuple(self.cu.iter_DIE_children(self))
+
     def iter_children(self):
-        """ Yield all children of this DIE
+        """ Iterates all children of this DIE
+
+            If you sure that all children will be processed then better
+            iterate `children` attribute.
         """
-        return iter(self._children)
+        try:
+            # Try to use lazy `children`.
+            return iter(self.__dict__["children"])
```

Comment:
If you have to do this manually anyway, did the `@lazy` decorator help here really?

Reply:
Lazy `children` attribute is mainly added for a user who wants exactly all children of a DIE.
This code serves to another user who wants only few children of a DIE. Because `iter_DIE_children` is slower that a `tuple` iteration, `iter_children` first tries to use lazily constructed `children` but does not forces its construction. It's an optimization only.